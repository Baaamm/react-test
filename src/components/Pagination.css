:local(.nextButton){
  float: right;
}

:local(.prevButton){
  float: left;
}
