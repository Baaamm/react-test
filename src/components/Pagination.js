import React from 'react';
import { PropTypes } from 'prop-types'

import styles from './Pagination.css';

const Pagination = ( { actualPage, totalPages, updatePage} ) => (
    <div>
        {actualPage > 1 &&
            <button className={`btn btn-default ${styles.prevButton}`}
                    onClick={() => updatePage(actualPage - 1)}>
                <span>pre</span>
            </button>
        }

        {actualPage < totalPages &&
            <button className={`btn btn-default ${styles.nextButton}`}
                    onClick={() => updatePage(actualPage + 1)}>
                <span>next</span>
            </button>
        }
    </div>
);

Pagination.propTypes = {
    actualPage: PropTypes.number.isRequired,
    totalPages: PropTypes.number.isRequired,
    updatePage: PropTypes.func.isRequired
};

export default Pagination;
