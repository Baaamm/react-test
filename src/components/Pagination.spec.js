import React from 'react';
import Pagination from './Pagination';
import renderer from 'react-test-renderer';

it('renders only with next button', () => {
    const tree = renderer.create(<Pagination actualPage={1} totalPages = {2} updatePage={null} />).toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders with next and prev button', () => {
    const tree = renderer.create(<Pagination actualPage={2} totalPages = {5} updatePage={null} />).toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders with only prev button', () => {
    const tree = renderer.create(<Pagination actualPage={5} totalPages = {5} updatePage={null} />).toJSON();
    expect(tree).toMatchSnapshot();
});
