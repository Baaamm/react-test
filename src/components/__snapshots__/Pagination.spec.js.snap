exports[`test renders only with next button 1`] = `
<div>
  <button
    className="btn btn-default undefined"
    onClick={[Function]}>
    <span>
      next
    </span>
  </button>
</div>
`;

exports[`test renders with next and prev button 1`] = `
<div>
  <button
    className="btn btn-default undefined"
    onClick={[Function]}>
    <span>
      pre
    </span>
  </button>
  <button
    className="btn btn-default undefined"
    onClick={[Function]}>
    <span>
      next
    </span>
  </button>
</div>
`;

exports[`test renders with only prev button 1`] = `
<div>
  <button
    className="btn btn-default undefined"
    onClick={[Function]}>
    <span>
      pre
    </span>
  </button>
</div>
`;
